src/Shared/Executor.cs

```diff
@@ -98,8 +104,31 @@ static MethodInfo getEntryPoint(MethodInfo main)
                         (byte)ILOpCode.Call, _, _, _, _,
                         (byte)ILOpCode.Ret
                     ] => main.Module.ResolveMethod(BitConverter.ToInt32(bytes.AsSpan(1, 4))),
+                    // Patterns for async Script Main method
+                    [
+                        (byte)ILOpCode.Newobj, _, _, _, _,
+                        (byte)ILOpCode.Callvirt, _, _, _, _,
+                        (byte)ILOpCode.Callvirt, _, _, _, _,
+                        (byte)ILOpCode.Stloc_0,
+                        (byte)ILOpCode.Ldloca_s, 0,
+                        (byte)ILOpCode.Call, _, _, _, _,
+                        (byte)ILOpCode.Pop,
+                        (byte)ILOpCode.Ret
+                    ] => CreateScriptMain(main, bytes),
                     _ => null,
                 };
+                static DynamicMethod? CreateScriptMain(MethodInfo main, byte[] bytes)
+                {
+                    ConstructorInfo? constructor = main.Module.ResolveMethod(BitConverter.ToInt32(bytes.AsSpan(1, 4))) as ConstructorInfo;
+                    MethodInfo? initialize = main.Module.ResolveMethod(BitConverter.ToInt32(bytes.AsSpan(6, 4))) as MethodInfo;
+                    if (constructor == null || initialize == null) { return null; }
+                    DynamicMethod method = new(main.Name, initialize.ReturnType, [], main.Module);
+                    ILGenerator generator = method.GetILGenerator();
```

Comment:
Emitting a dynamic method seems like an overkill, couldn't we instead instantiate the script class via reflection and returns its initialize method? (May need some support in the caller to invoke instance method, but that seems fine.)

Reply:
> In fact it is a reflection.

Sure, but it feels to me that creating a DynamicMethod, having it JIT'd, just to be invoked once, is a very heavy-weight kind of reflection.

> Or we can create a delegate:

Yes, I think that's better. I can make the change, thanks.

Reply:
![😕](https://github.com/user-attachments/assets/bb59740e-7fb4-47b1-8f0f-1f89a517b7ab)